app/bin/service/search.dart

```diff
@@ -23,7 +23,23 @@ import 'package:pub_dartlang_org/search/handlers.dart';
 import 'package:pub_dartlang_org/search/index_simple.dart';
 import 'package:pub_dartlang_org/search/updater.dart';
 
-void main() {
+Future main() async {
+  final ReceivePort errorPort = new ReceivePort();
+  errorPort.listen((error) {
+    print('ERROR from isolate: $error');
+  });
+  for (int i = 0; i < envConfig.isolateCount; i++) {
+    await Isolate.spawn(
+      _main,
+      i,
+      onError: errorPort.sendPort,
+      onExit: errorPort.sendPort,
+      errorsAreFatal: true,
+    );
+  }
+}
+
+void _main(int isolateId) {
```

Comment:
If we have several isolate running the normal search indexing & serving code, they will also race when writing checkpoints of the index and deleting old checkpoints. But I think it should be guaranteed that there's always one checkpoint left.